### DISCUSSION TITLE
Potential memory allocation issues on WASM

### DISCUSSION THREAD

**Discussion by maxammann** [maplibre/maplibre-rs#83] [General]
Just found this major issue: https://github.com/WebAssembly/design/issues/1397

It seems like allocating memory on WASM is problematic if we do not know the heap consumption at compile time.
I think we could work around for the default usecase. We could calculate some worst-case heap usage.

This is frightening nonetheless...

@bchapuis this could be interesting for the paper

**Comment by bchapuis** [comment]
That's really interesting, thank you for the link. Have you encountered this issue in maplibre-rs already? I looks like a major concern on chrome for mobile where allocating more than ~300MB of memory is not reliable.

**Reply by maxammann** [reply]
Not sofar. But I'm now wondering for which kind of applications 300MB are hit. For example is raster tile rendering even possible with WASM?
